java/gateway-core/src/main/java/com/avaulta/gateway/rules/Endpoint.java

```diff
@@ -91,12 +91,25 @@ public Optional<Map<String, QueryParameterSchema>> getQueryParamSchemasOptional(
      *
      * if omitted, any HTTP method is permitted.
      */
-    @JsonInclude(JsonInclude.Include.NON_EMPTY)
     Set<String> allowedMethods;
 
+    @JsonInclude(JsonInclude.Include.NON_EMPTY)
+    public Set<String> getAllowedMethods() {
+        // our own implementation, so consistently sorted.
+        if (allowedMethods == null) {
+            return null;
+        } else if (allowedMethods instanceof TreeSet) {
+            return allowedMethods;
+        } else {
+            //rely on natural ordering of strings to sort
+            this.allowedMethods = new TreeSet<>(allowedMethods);
+            return this.allowedMethods;
+        }
+    }
```

Comment:
@aperez-worklytics  FYI; above done bc tests in endpoint serialization passing locally for me, but ordering was different in CI.  This should make it consistent.